src/backend/distributed/master/master_modify_multiple_shards.c

```diff
@@ -0,0 +1,403 @@
+		uint32 workerPort = shardPlacement->nodePort;
+		char *nodeUser = CurrentUserName();
+		PGconn *connection = ConnectToNode(workerName, workerPort, nodeUser);
+		TransactionConnection *transactionConnection = NULL;
+
+		if (connection == NULL)
+		{
+			List *abortConnectionList = ConnectionList(shardConnectionHash);
+			CloseConnections(abortConnectionList);
+
+			ereport(ERROR, (errmsg("Could not establish a connection to all "
+								   "placements.")));
+		}
+
+		transactionConnection = palloc0(sizeof(TransactionConnection));
+
+		transactionConnection->connectionId = shardConnections->shardId;
+		transactionConnection->transactionState = TRANSACTION_STATE_INVALID;
+		transactionConnection->connection = connection;
+
+		connectionList = lappend(connectionList, transactionConnection);
+	}
+
+	shardConnections->connectionList = connectionList;
+}
+
+
+/*
+ * SendQueryToPlacement sends the given query string to all given placement
+ * connections of a shard. The query is sent with a BEGIN before the the actual
+ * query so, CommitRemoteTransactions or AbortRemoteTransactions should be
+ * called after all queries have been sent successfully.
+ */
+static int
+SendQueryToConnections(char *shardQueryString, ShardConnections *shardConnections)
+{
+	uint64 shardId = shardConnections->shardId;
+	List *connectionList = shardConnections->connectionList;
+	ListCell *connectionCell = NULL;
+	int32 shardAffectedTupleCount = -1;
+
+	Assert(connectionList != NIL);
+
+	foreach(connectionCell, connectionList)
+	{
+		TransactionConnection *transactionConnection =
+			(TransactionConnection *) lfirst(connectionCell);
+		PGconn *connection = transactionConnection->connection;
+		PGresult *result = NULL;
+		char *placementAffectedTupleString = NULL;
+		int32 placementAffectedTupleCount = -1;
+
+		/* send the query */
+		result = PQexec(connection, "BEGIN");
+		if (PQresultStatus(result) != PGRES_COMMAND_OK)
+		{
+			ReportRemoteError(connection, result);
+			ereport(ERROR, (errmsg("Could not send query to shard placement.")));
+		}
+
+		result = PQexec(connection, shardQueryString);
+		if (PQresultStatus(result) != PGRES_COMMAND_OK)
+		{
+			ReportRemoteError(connection, result);
+			ereport(ERROR, (errmsg("Could not send query to shard placement.")));
+		}
+
+		placementAffectedTupleString = PQcmdTuples(result);
+		placementAffectedTupleCount = pg_atoi(placementAffectedTupleString,
+											  sizeof(int32), 0);
+
+		if ((shardAffectedTupleCount == -1) ||
+			(shardAffectedTupleCount == placementAffectedTupleCount))
+		{
+			shardAffectedTupleCount = placementAffectedTupleCount;
+		}
+		else
+		{
+			ereport(ERROR,
+					(errmsg("modified %d tuples, but expected to modify %d",
+							placementAffectedTupleCount, shardAffectedTupleCount),
+					 errdetail("Affected tuple counts at placements of shard "
+							   UINT64_FORMAT " are different.", shardId)));
+
+			if (placementAffectedTupleCount > shardAffectedTupleCount)
```

Comment:
Corrected. After changing the report type from WARNING to ERROR, I forgot to remove that part.